perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/AppAutoRejectionScheduler.java

```diff
@@ -174,43 +197,154 @@ private void rejectExpiredApplications (List<Application> applications, Attribut
+	 * such attribute set, the default message will be used.
 	 *
-	 * @param application application to reject
-	 * @param reasonForVo reason for reject VO application
-	 * @param reasonForGroup reason for reject group application
+	 * @param application application to be rejected
+	 * @param attrValueKey key of a message in the applicationAutoRejectMessages
+	 * @param defaultMessage default message if the appropriate vo/group has no
+	 *                       applicationAutoRejectMessages attribute set
 	 */
-	private void rejectWithReason (Application application, String reasonForVo, String reasonForGroup) {
+	private void rejectApplication(Application application, String attrValueKey, String defaultMessage) {
+		Group group = application.getGroup();
+		String lang = getUserPreferredLang(application);
+		Vo vo = application.getVo();
+
+		Attribute messagesAttr = null;
 		try {
-			if (application.getGroup() == null) {
-				registrarManager.rejectApplication(sess, application.getId(), reasonForVo);
+			if (group != null) {
+				messagesAttr = perun.getAttributesManagerBl().getAttribute(sess, group, A_GROUP_APP_REJECT_MESSAGES);
 			} else {
-				registrarManager.rejectApplication(sess, application.getId(), reasonForGroup);
+				messagesAttr = perun.getAttributesManagerBl().getAttribute(sess, vo, A_VO_APP_REJECT_MESSAGES);
 			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Failed to get attribute with reject messages.", e);
+		}
+
+		String message;
+		if (group != null) {
+			message = getGroupMessage(messagesAttr, attrValueKey, lang, defaultMessage, group);
+		} else {
+			message = getVoMessage(messagesAttr, attrValueKey, lang, defaultMessage, vo);
+		}
+
+		try {
+			registrarManager.rejectApplication(sess, application.getId(), message);
 		} catch (PerunException e) {
-			log.error("Failed to reject expired application: {}", application, e);
+			log.error("Failed to reject application {}.", application, e);
+		}
+	}
+
+
+	/**
+	 * Returns reject message obtained from the given attribute, using the given key; or returns the default msg,
+	 * if there is no suitable message in the attribute. If there is a %group_name% placeholder, it will
+	 * be replaced by the given group's name.
+	 *
+	 * @param attribute attribute of type map
+	 * @param lang user preferred language
+	 * @param key key used to find the value in the given attribute
+	 * @param defaultMsg returned if there is no suitable message in the attribute
+	 * @param group this group's name will be placed in the message, if there is a %group_name% placeholder
+	 * @return reject message, concerning the preferred language, obtained from the given attribute,
+	 *         or the given default message
+	 */
+	private String getGroupMessage(Attribute attribute, String key, String lang, String defaultMsg, Group group) {
+		return getRejectMessage(attribute, lang, key, defaultMsg)
+				.replaceAll(GROUP_PLACEHOLDER, group.getName());
+	}
+
```

Comment:
Is it really necessary to create method just for this one-liner?

Reply:
It probably isn't. I will refactor it.